cli/azd/cmd/middleware/experimentation.go

```diff
@@ -0,0 +1,65 @@
+package middleware
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/cmd/actions"
+	"github.com/azure/azure-dev/cli/azd/internal/tracing"
+	"github.com/azure/azure-dev/cli/azd/internal/tracing/fields"
+	"github.com/azure/azure-dev/cli/azd/pkg/alpha"
+	"github.com/azure/azure-dev/cli/azd/pkg/experimentation"
+)
+
+type ExperimentationMiddleware struct {
+}
+
+func NewExperimentationMiddleware() Middleware {
+	return &ExperimentationMiddleware{}
+}
+
+const assignmentEndpoint = "https://default.exp-tas.com/exptas49/b80dfe81-554e-48ec-a7bc-1dd773cd6a54-azdexpws/api/v1/tas"
+
+func (m *ExperimentationMiddleware) Run(ctx context.Context, next NextFn) (*actions.ActionResult, error) {
+	endpoint := assignmentEndpoint
+	// Allow overriding the assignment endpoint, either for local development (where you want to hit a private instance)
+	// or testing (we use this in our end to end tests to control assignment behavior for the CLI under test)/
+	if override := os.Getenv("AZD_DEBUG_EXPERIMENTATION_TAS_ENDPOINT"); override != "" {
+		log.Printf("using override assignment endpoint: %s, from AZD_DEBUG_EXPERIMENTATION_TAS_ENDPOINT", override)
+		endpoint = override
+	}
+
+	if assignmentManager, err := experimentation.NewAssignmentsManager(
```

Comment:
Consider moving this manager instantiation to the IoC container and passed in via DI to be consistent with other components.

Reply:
Will separate out as a separate change. Right now, it's a silent failure if the assignment manager fails. I'm not sure what we need to do to replicate that behavior in DI. Potentially using `Lazy` but that's a misnomer, we want silent failure handling, not necessarily lazy as-is instantiation. Will need to think about it a little.